src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java

```diff
@@ -0,0 +1,136 @@
+package org.spongepowered.common.data.processor.data;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.pathfinding.PathPoint;
+import org.spongepowered.api.data.*;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.ImmutableTargetedLocationData;
+import org.spongepowered.api.data.manipulator.mutable.TargetedLocationData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.Sponge;
+import org.spongepowered.common.data.DataProcessor;
+import org.spongepowered.common.data.manipulator.immutable.ImmutableSpongeTargetedLocationData;
+import org.spongepowered.common.data.manipulator.mutable.SpongeTargetedLocationData;
+import org.spongepowered.common.data.util.DataUtil;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class TargetedLocationDataProcessor implements DataProcessor<TargetedLocationData, ImmutableTargetedLocationData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<TargetedLocationData> from(DataHolder dataHolder) {
+        if(dataHolder instanceof EntityLiving) {
```

Comment:
@JBYoshi If you have multiple compasses, they use the "respawn" location for the player, however, we can use the `TargetedLocationData` as I've been suggesting for a while now.